src/include/migraphx/op/deconvolution.hpp

```diff
@@ -37,27 +37,41 @@ struct deconvolution
     }
 
     std::string name() const { return "deconvolution"; }
+
+    void check_attribute_size() const
+    {
+        if(not(padding.size() == stride.size() and padding.size() == dilation.size()))
+        {
+            MIGRAPHX_THROW("deconvolution: inconsistent attribute sizes");
+        }
+    }
+
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.has(2).same_type().same_ndims().only_dims(4);
+        check_shapes{inputs, *this}.has(2).same_type().same_ndims().min_ndims(3);
+        check_attribute_size();
 
```

Comment:
We need a check to ensure input dim num and attribute dim num match. This is also needed for the convolution. I spend two ours figure out a bug that input dim number is different from the attribute dim number since we did not report such a wrong input scenario.

Reply:
fixed